MobileApplication/MobileApplication/ViewModels/Category/CategoryViewModel.cs

```diff
@@ -9,42 +9,95 @@
+using System.Collections.Generic;
 
 namespace MobileApplication.ViewModels
 {
     public class CategoryViewModel : BaseViewModel
     {
         private Category _selectedItem;
 
-        public ObservableCollection<Category> Items { get; }
+        private List<Category> _items { get; set; }
+        public ObservableCollection<CategoryView> CategoryViews { get; }
+        private List<Transaction> _transactions { get; }
         public Command LoadItemsCommand { get; }
         public Command AddItemCommand { get; }
         public Command<Category> ItemTapped { get; }
 
         private readonly IRestService<Category> RestService;
+        private readonly IRestService<Transaction> TransactionRestService;
 
         public CategoryViewModel()
         {
             RestService = new RestService<Category>("api/Categories");
+            TransactionRestService = new RestService<Transaction>("api/Transactions");
             Title = "Categories";
-            Items = new ObservableCollection<Category>();
+       
+            CategoryViews = new ObservableCollection<CategoryView>();
             LoadItemsCommand = new Command(async () => await ExecuteLoadItemsCommand());
 
             ItemTapped = new Command<Category>(OnItemSelected);
 
             AddItemCommand = new Command(OnAddItem);
         }
+        //public void Calculation(List<Category> items, List<Transaction> transactions)
+        //{
+        //    foreach (var category in items)
+        //    {
+        //        decimal calc = 0;
+        //        foreach (var transaction in transactions)
+        //        {
+        //            if (category.Title == transaction.Category.Title)
+        //            {
+        //                calc += transaction.Amount;
+        //            }
+        //        }
 
-        async Task ExecuteLoadItemsCommand()
+        //        var newcategoryview = new CategoryView()
+        //        {
+        //            Title = category.Title,
+        //            DedicatedAmount = category.DedicatedAmount,
+        //            BudgetedAmount = Math.Abs(calc),
+        //            AvailableAmount = category.DedicatedAmount - Math.Abs(calc)
+        //        };
+
+        //        CategoryViews.Add(newcategoryview);
+                
+        //    }
+        //}
+        public async Task ExecuteLoadItemsCommand()
         {
             IsBusy = true;
-            Items.Clear();
+            CategoryViews.Clear();
 
             try
             {
                 var items = await RestService.RefreshDataAsync();
+                var transactions = await TransactionRestService.RefreshDataAsync();
+
+                foreach (var category in items)
+                {
+                    decimal calc = 0;
+                    foreach (var transaction in transactions)
+                    {
+
+                        if (transaction.Category != null && category.Title == transaction.Category.Title)
```

Comment:
Why do you check if it's null if you also compare the title?

Reply:
Because if its null it will catch null reference exception.